### DISCUSSION TITLE
Example use of transactions in async method

### DISCUSSION THREAD

**Discussion by bsiewert-aya** [mysql-net/MySqlConnector#1531] [Q&A] [answered]
Hello, 

I'm new to MySqlConnector and had a question about the use of transactions. 

I'm studying the example console application in the documentation found here: 
https://mysqlconnector.net/tutorials/best-practices/

With this example below, can you provide a proper usage with transactions if I were to use the same task in the example console app but use a transaction to upsert records in each task? 

![image](https://github.com/user-attachments/assets/d23cd1b4-b260-4c11-8347-448a37e0d4cf)

**Comment by bgrainger** [comment, marked as answer]
You would use [BeginTransactionAsync](https://mysqlconnector.net/api/mysqlconnector/mysqlconnection/begintransactionasync/) and [CommitAsync](https://mysqlconnector.net/api/mysqlconnector/mysqltransaction/commitasync/).
